beacon-chain/rpc/proposer_server.go

```diff
@@ -110,29 +111,17 @@ func (ps *ProposerServer) PendingAttestations(ctx context.Context, req *pb.Pendi
 		currentSlot = beaconState.Slot
 	}
 
-	// Remove any attestation from the list if their slot is before the start of
-	// the previous epoch or does not match the current state previous justified
-	// epoch or attestation is not voting on the canonical chain.
-	// This should be handled in the operationService cleanup but we
-	// should filter here in case it wasn't yet processed.
-	boundary := currentSlot - params.BeaconConfig().SlotsPerEpoch
-	attsWithinBoundary := make([]*pbp2p.Attestation, 0, len(atts))
+	validAtts := make([]*pbp2p.Attestation, 0, len(atts))
 	for _, att := range atts {
-
-		var expectedJustifedEpoch uint64
-		if helpers.SlotToEpoch(att.Data.Slot+1) >= helpers.SlotToEpoch(currentSlot) {
-			expectedJustifedEpoch = beaconState.JustifiedEpoch
-		} else {
-			expectedJustifedEpoch = beaconState.PreviousJustifiedEpoch
+		if err := blocks.VerifyAttestation(beaconState, att, false); err != nil {
+			log.WithField("slot", att.Data.Slot-
+				params.BeaconConfig().GenesisSlot).Warn("Skipping, pending attestation failed verification")
```

Comment:
Do you want to log WithError(err)?

Reply:
yeah... error feels more appropriate here